custom_components/foxess_modbus/entities/modbus_entity_mixin.py

```diff
@@ -74,3 +67,29 @@ def update_callback(self, changed_addresses: set[int]) -> None:
         """Schedule a state update."""
         if self.entity_description.address in changed_addresses:
             self.schedule_update_ha_state(True)
+
+    def _get_unique_id(self):
+        """Get unique ID"""
+        friendly_name = self._inv_details[FRIENDLY_NAME]
+        if friendly_name != "":
+            return f"{friendly_name}_{self.entity_description.key}"
+        else:
+            return f"{self.entity_description.key}"
+
+    def _validate(self, rules: list[BaseValidator], processed, original=None) -> bool:
+        """Validate against a set of rules"""
+        original = original if not None else processed
```

Comment:
`original = original if original is not None else processed`.

Currently this is the same as `original = original`, since `not None` is always True.

Reply:
:)